app/gui2/src/components/ComponentBrowser.vue

```diff
@@ -261,6 +262,18 @@ const previewedExpression = debouncedGetter(() => {
   else return input.inputAfterApplyingSuggestion(selectedSuggestion.value).newCode
 }, 200)
 
+const previewedExpressionSuggestionId = ref<SuggestionId>()
+
+watch(previewedExpression, () => {
+  previewedExpressionSuggestionId.value = selectedSuggestionId.value ?? undefined
+})
```

Comment:
~~Why isn't `previewedExpressionSuggestionId` a `computed` ref?~~, after a bit of thinking I know. But add a short comment explaining it, as it's not very obvious.

Or, make `previewedExpression` a pair of `string` and optional `SuggestionId` - the fewer watches the better.

Also, `Expression` part in this and similar names is redundant IMO, it could be just `previewedSuggestionId`.